api/external/maistra/status/status.go

```diff
@@ -0,0 +1,279 @@
+package status
```

Comment:
why do we need this code in the kuadrant operator repo?
why not import them?

Reply:
I'm afraid the types cannot be imported directly.

The packages have deep dependencies that are based on old golang version and libraries not compatible with the version of our code. Luckily we don't depend on those incompatible bit as we only need the type definitions.

For the record, this wasn't my first option either, but it turned out to be the best approach until Maistra team upgrades to newer version of Operator SDK, something I was told to be in their roadmap.

BTW, the approach of hosting copies the types is adopted by https://github.com/maistra/istio-operator itself.